networkx/algorithms/components/weakly_connected.py

```diff
@@ -83,6 +83,14 @@ def number_weakly_connected_components(G):
     NetworkXNotImplemented
         If G is undirected.
 
+    Examples
+    --------
+    >>> import networkx as nx
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from([(0, 1), (2, 1), (3, 4)])
```

Comment:
IMO it's better to keep the example as simple as possible, so we can get rid of the explicit `add_edges_from` call:

```suggestion
    >>> G = nx.DiGraph([(0, 1), (2, 1), (3, 4)])
```